src/editor/Collaboration.js

```diff
@@ -0,0 +1,109 @@
+define(function (require, exports, module) {
+    "use strict";
+
+    var SimpleWebRTC    = require("simplewebrtc");
+
+    function Collaboration() {
+        var webrtc = new SimpleWebRTC({
+            // the id/element dom element that will hold "our" videos
+            localVideoEl: 'localVideo',
+            // the id/element dom element that will hold remote videos
+            remoteVideosEl: 'remotesVideos',
+            // immediately ask for camera access
+            autoRequestMedia: false
+        });
+        this.webrtc = webrtc;
+        this.pending = []; // pending clients that need to be initialized.
+        this.changing = false;
+    };
+
+    Collaboration.prototype.init = function(codemirror) {
+        var self = this;
+        this.webrtc.joinRoom('thimble', function() {
+            self.codemirror = codemirror;
+            self.webrtc.sendToAll("new client", {});
+            self.addListeners();
+        });
+    };
+
+    Collaboration.prototype.addListeners = function() {
```

Comment:
Move this code inline in `init` above vs. putting it on the prototype, since it's never called anywhere else.